frontend/src/routes/Infrastructure/InfraEnvironments/InfraEnvironmentsPage.tsx

```diff
@@ -269,19 +309,81 @@ const InfraEnvsTable: React.FC<InfraEnvsTableProps> = ({ infraEnvs, agents }) =>
                                         sort: 'metadata.name',
                                     },
                                     {
-                                        header: t('infraEnv.tableHeader.namespace'),
-                                        cell: 'metadata.namespace',
-                                        sort: 'metadata.namespace',
+                                        header: t('infraEnv.tableHeader.hosts'),
+                                        cell: (infraEnv) => {
+                                            const infraAgents = agents.filter((a) =>
+                                                isMatch(
+                                                    a.metadata.labels,
+                                                    infraEnv.status?.agentLabelSelector?.matchLabels
+                                                )
+                                            )
+                                            const errorAgents = infraAgents.filter(
+                                                (a) => getAgentStatus(a).status.key === 'error'
+                                            )
+                                            const warningAgents = infraAgents.filter((a) =>
+                                                ['pending-for-input', 'insufficient', 'insufficient-unbound'].includes(
+                                                    getAgentStatus(a).status.key
+                                                )
+                                            )
+                                            return (
+                                                <Link to={`${getDetailsLink(infraEnv)}/hosts`}>
```

Comment:
This needs to be moved to wrap content on line 333 otherwise the popover on 'cannot be deleted' does not work.